tests/phpunit/tests/block-template.php

```diff
@@ -275,6 +275,83 @@ public function test_get_the_block_template_html_does_not_generally_enforce_loop
 		$this->assertSame( array( false, true ), $in_the_loop_logs, 'Main query loop was triggered incorrectly' );
 	}
 
+	/**
+	 * @ticket 58319
+	 *
+	 * @covers ::get_block_theme_folders
+	 * @dataProvider data_get_block_theme_folders
+	 */
+	public function test_get_block_theme_folders( $theme, $expected, $message ) {
+		$this->assertSame( $expected, get_block_theme_folders( $theme ), $message );
+
+		// Test that the result is cached.
+		if ( empty( $theme ) ) {
+			$theme = get_stylesheet();
+		}
+		$cache_key    = 'theme-' . md5( get_theme_root( $theme ) . '/' . $theme );
+		$theme_cache  = wp_cache_get( $cache_key, 'themes' );
+		$cached_value = $theme_cache['block_template_folders'];
+		// Test for value stored in cache.
+		wp_cache_delete( $cache_key ); // clean up for next test.
+		$this->assertSame( $expected, $cached_value, $message . ' to be cached' );
```

Comment:
```suggestion
		$this->assertSame( $expected, $cached_value, $message );
```

Let's just add `to be cached'` to the end of all the messages. It is cleaner.

Reply:
@spacedmonkey I was trying to reuse same message for both the assertion with some minor changes to differentiate in case of failed assertion. I have remove the postfix and introduced a prefix "Cache - ".